File: frontend/src/types/insights.ts
export interface CountrySalaryStats {
  country: string;
  min_salary: number;
  max_salary: number;
  avg_salary: number;
  employee_count: number;
}

export interface JobTitleSalaryStats {
  job_title: string;
  country: string;
  avg_salary: number;
  employee_count: number;
}

export interface CountryHeadcount {
  country: string;
  employee_count: number;
}

export interface DepartmentSummary {
  department: string;
  avg_salary: number;
  min_salary: number;
  max_salary: number;
  employee_count: number;
}

export interface TopEarner {
  id: string;
  full_name: string;
  job_title: string;
  country: string;
  department: string;
  salary: number;
}

File: backend/app/schemas/insights.py
from decimal import Decimal

from pydantic import BaseModel, ConfigDict


class CountrySalaryStats(BaseModel):
    country: str
    min_salary: float
    max_salary: float
    avg_salary: float
    employee_count: int


class JobTitleSalaryStats(BaseModel):
    job_title: str
    country: str
    avg_salary: float
    employee_count: int


class CountryHeadcount(BaseModel):
    country: str
    employee_count: int


class DepartmentSummary(BaseModel):
    department: str
    avg_salary: float
    min_salary: float
    max_salary: float
    employee_count: int


class TopEarner(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    full_name: str
    job_title: str
    country: str
    department: str
    salary: Decimal
